Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:14.04
+FROM ubuntu:16.04
```

Comment:
I worry a little that this may cause issues with some of the other scenarios, so it would be best if we could limit the scope of this change to gzip. The simplest way to do that is just to copy the contents of this file, append it to the top of the Dockerfile for gzip, and then remove the existing `FROM ...` statement in the gzip Dockerfile.

Reply:
On it, boss.

Reply:
Sooooo, hit a minor snag.  The contents of `base` dir is referenced from the top level Dockerfile, which does not exist at the gzip directory structure.  I'm resolving this by duplicating the base directory in gzip dir. I hate duplication, but the `base` file is out of scope with the gzip Dockerfile level.  same for compile.sh.